subworkflows/local/metagenomics_profiling.nf

```diff
@@ -137,20 +137,32 @@ workflow METAGENOMICS_PROFILING {
     }
 
     else if ( params.metagenomics_profiling_tool == 'krakenuniq' ) {
-        // run krakenuniq once for all samples
+        // run krakenuniq once for all samples, unless non-merged PE vs SE data
 
         ch_krakenuniq_input = ch_reads
             .map{ meta, file ->
                 [
```

Comment:
if the meta already contains the informatino, this whole map can go away. I created this map only to add the 'single_end' back to the meta

Reply:
but we do want to have 2 groups of files for input into krakenuniq (which runs all samples in a single execution): one single_end group and one paired end group. Therefore for groupTuple() we need to have a new meta-like field (not in meta explicitly) that keeps just the SE vs PE info -> sort

Reply:
Ah, I see. Alright, then this makes sense :)